core/src/main/scala/org/http4s/jdkhttpclient/JdkHttpClient.scala

```diff
@@ -243,10 +243,28 @@ object JdkHttpClient {
     * [[cats.effect.kernel.Async.executor executor]], sets the
     * [[org.http4s.client.defaults.ConnectTimeout default http4s connect timeout]], and disables
     * [[https://github.com/http4s/http4s-jdk-http-client/issues/200 TLS 1.3 on JDK 11]].
+    *
+    * On Java 21 and higher, prefer [[simpleResource]] as it actively closes the underlying client,
+    * releasing its resources early.
     */
   def simple[F[_]](implicit F: Async[F]): F[Client[F]] =
     defaultHttpClient[F].map(apply(_))
 
+  /** Like [[simple]], but wraps the client in a [[cats.effect.Resource Resource]] to ensure it's
+    * properly shut down on JVM 21 and higher. On lower Java versions, closing the resource does
+    * nothing (garbage collection will eventually clean up the client).
+    */
+  def simpleResource[F[_]](implicit F: Async[F]): Resource[F, Client[F]] =
```

Comment:
Name is up for discussion, I don't particular like `simpleResource`, but at least it's rather clear.

Reply:
As already mentioned in https://github.com/http4s/http4s-jdk-http-client/issues/1011#issuecomment-1919676958, we could just change the signature of `simple` and cut a 0.10 release; it should be rather simple for users to migrate.

Reply:
I think we should rename `simple` and `simpleResource` to something like `unsafeDefault` and `default`. We should encourage users to do the right thing (using the resource variant)

We can do that in a 0.10 release, or deprecate simple in this and create new new constructors.

Reply:
@hamnis I can adjust the names, no problem, just a bit unsure about the `unsafe` prefix. On everything below Java 21, it's actually not unsafe in the sense CE usually uses this, but the only thing you can have (the `Resource` variant won't close anything there either). 
So maybe a 0.10 with `simple` (returning a `Resource`, by re-using the name users get nudged towards this) and a `def forEffect[F[_]: Async]: F[Client[F]]` (or any better name) with docs explaining the caveats on Java 21+?

Reply:
do we really need the forEffect constructor? Do we lose anything with just having the `simple` | `default` | `insertyournamehere` resource constructor ?

Reply:
@hamnis It may cause some (unnecessary) trouble for users who now need to bubble up `Resource` usage in case they hadn't so far. But as `allocated` is safe on Java below 21, I changed the PR to only offer `def simple[F[_]: Async]: Resource[F, …]` and mentioned `allocated` in the docs, so people have a hint on a workaround. Hope that's a fair compromise!

Reply:
Obviously, that means MiMa is unhappy and we need to release it as `0.10`.